test/test_chapeldomain.py

```diff
@@ -766,51 +767,56 @@ def test_with_prefixes(self):
+            ('proc foo() where a > b', 'proc ', None, 'foo', '', None, ' where a > b'), 
+            ('proc foo:int where a > b', 'proc ', None, 'foo', None, ':int', ' where a > b'), 
+            ('proc foo():int where a > b', 'proc ', None, 'foo', '', ':int', ' where a > b'), 
+            ('proc foo ref where a > b', 'proc ', None, 'foo', None, ' ref', ' where a > b'), 
+            ('proc foo() ref where a > b', 'proc ', None, 'foo', '', ' ref', ' where a > b'), 
+            ('proc foo ref: int where a > b', 'proc ', None, 'foo', None, ' ref: int', ' where a > b'), 
+            ('proc foo() ref: int where a > b', 'proc ', None, 'foo', '', ' ref: int', ' where a > b'),
+            ('proc foo() ref', 'proc ', None, 'foo', '', ' ref', None),
+            ('iter foo() ref', 'iter ', None, 'foo', '', ' ref', None),
+            ('inline proc Vector.pop() ref', 'inline proc ', 'Vector.', 'pop', '', ' ref', None),
+            ('inline proc range.first', 'inline proc ', 'range.', 'first', None, None, None),
+            ('proc My.Mod.With.Deep.NameSpace.1.2.3.432.foo()', 'proc ', 'My.Mod.With.Deep.NameSpace.1.2.3.432.', 'foo', '', None, None),
+            ('these() ref', None, None, 'these', '', ' ref', None),
+            ('size', None, None, 'size', None, None, None),
+            ('proc Util.toVector(type eltType, cap=4, offset=0): Containers.Vector', 'proc ', 'Util.', 'toVector', 'type eltType, cap=4, offset=0', ': Containers.Vector', None),
+            ('proc MyClass$.lock$(combo$): sync bool', 'proc ', 'MyClass$.', 'lock$', 'combo$', ': sync bool', None),
+            ('proc MyClass$.lock$(combo$): sync myBool$', 'proc ', 'MyClass$.', 'lock$', 'combo$', ': sync myBool$', None),
+            ('proc type currentTime(): int(64)', 'proc type ', None, 'currentTime', '', ': int(64)', None),
+            ('proc param int.someNum(): int(64)', 'proc param ', 'int.', 'someNum', '', ': int(64)', None),
+            ('proc MyRs(seed: int(64)): int(64)', 'proc ', None, 'MyRs', 'seed: int(64)', ': int(64)', None),
             ('proc RandomStream(seed: int(64) = SeedGenerator.currentTime, param parSafe: bool = true)',
-             'proc ', None, 'RandomStream', 'seed: int(64) = SeedGenerator.currentTime, param parSafe: bool = true', None),
-            ('class X', 'class ', None, 'X', None, None),
-            ('class MyClass:YourClass', 'class ', None, 'MyClass', None, ':YourClass'),
-            ('class M.C : A, B, C', 'class ', 'M.', 'C', None, ': A, B, C'),
-            ('record R', 'record ', None, 'R', None, None),
-            ('record MyRec:SuRec', 'record ', None, 'MyRec', None, ':SuRec'),
-            ('record N.R : X, Y, Z', 'record ', 'N.', 'R', None, ': X, Y, Z'),
+             'proc ', None, 'RandomStream', 'seed: int(64) = SeedGenerator.currentTime, param parSafe: bool = true', None, None),
+            ('class X', 'class ', None, 'X', None, None, None),
```

Comment:
Hi Lydia, not sure what you mean by this. Do you mind elaborating?

Reply:
Sure!  Most of the checks here are on functions - procs, iters, operators.  Those all follow a set pattern that looks pretty similar.  Record and class signatures are a different sort of declaration, though I suppose the difference is more distinct in Chapel code than in the generated Sphinx signatures.  But because where clauses are only ever associated with functions, it definitely wouldn't make sense for a Sphinx record or class signature to include it, so it seems odd to me for them to even have a where clause entry.  The value in that case should always be None.

If we can't make it so that record and class signatures avoid having that argument in the group, it seems like it would be nice to have a check somewhere in the __init__.py file that will raise a ValueError or something if we know we've been given a record or class and the where_clause is set to something other than None.

Does that help clarify?  I'm happy to have a meeting to explain it better, too